com.ibm.wala.cast.python.ml/source/com/ibm/wala/cast/python/ml/client/PythonTensorAnalysisEngine.java

```diff
@@ -85,17 +91,43 @@ private static Set<PointsToSetVariable> getDataflowSources(Graph<PointsToSetVari
 									IR ir = node.getIR();
 									Value memberRefValue = ir.getSymbolTable().getValue(memberRef);
 
-									if (memberRefValue.isStringConstant()) {
-										tensorFlowAPI = ir.getSymbolTable().getStringValue(memberRef);
+									// while a member ref value exists and it's a string constant.
+									while (memberRefValue != null && memberRefValue.isStringConstant()) {
+										// push the API onto the queue.
+										tensorFlowAPIQueue.add(ir.getSymbolTable().getStringValue(memberRef));
+
+										// go back one "level."
+										memberRefValue = ir.getSymbolTable().getValue(--memberRef);
 									}
 								}
 							}
 						}
 					}
 
-					if ((ni.getCallSite().getDeclaredTarget().getName().toString().equals("read_data")
-							|| Objects.equal(tensorFlowAPI, "ones")) && ni.getException() != vn) {
-						sources.add(src);
+					// processs the API "levels."
+					if (ni.getException() != vn) {
+						if (!tensorFlowAPIQueue.isEmpty()) {
+							// pop the first element.
+							String tensorFlowAPI = tensorFlowAPIQueue.remove();
+
+							// Single-level APIs.
+							if (Objects.equal(tensorFlowAPI, "ones") || Objects.equal(tensorFlowAPI, "Variable")
```

Comment:
We have a test for `ones()`: https://github.com/wala/ML/blob/16cfbf4f04e52a343d97601c38a67ac60727b643/com.ibm.wala.cast.python.test/data/tf2b.py#L7
And, you've added a test for `random.uniform()` before. But, what about this other API mentioned here?

Reply:
I have added more tests.